### PR TITLE
Add pyscf_dispersion

### PR DESCRIPTION
**Pull request by wxj6000**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by wxj6000** [comment]
@conda-forge/python I was trying to skip win platform. It still raise error. Otherwise, it is ready for review.

**Comment by wxj6000** [comment]
@conda-forge/staged-recipes

**Comment by wxj6000** [comment]
@conda-forge/help-python-c I was trying to skip win platform. It still raise error. Otherwise, it is ready for review.

**Comment by wxj6000** [comment]
@conda-forge/help-python-c @conda-forge/staged-recipes
I was trying to skip win platform. It still raise error. Otherwise, it is ready for review.

**Comment by wxj6000** [comment]
@conda-forge/help-python

**Comment by wxj6000** [comment]
win_64 passed now. Ready for review.

File: recipes/pyscf-dispersion/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (xhochy) [inline recipes/pyscf-dispersion/meta.yaml:1]: Please remove these comments.

File: recipes/pyscf-dispersion/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - pyscf >=2.6
+  run:
+    - python
+    - pyscf >=2.6
+test:
+  imports:
+    - pyscf
+  requires:
+    - pip
+    - pyscf >=2.6
```
Comment (xhochy) [inline recipes/pyscf-dispersion/meta.yaml:50]: This should not be necessary.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/pyscf-dispersion/meta.yaml (hunk 2 above)
Comment (wxj6000) [reply recipes/pyscf-dispersion/meta.yaml:41]: It is updated with more accurate tests.

Comment (xhochy) [reply recipes/pyscf-dispersion/meta.yaml:41]: You should not need this depedency here, it's already in run.

File: recipes/pyscf-dispersion/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - {{ stdlib("c") }}
+    - cmake >=3.26
+    - make
+    - python
```
Comment (xhochy) [inline recipes/pyscf-dispersion/meta.yaml:24]: `python` in host should be enough
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**